contracts/EVMRuntime.sol

```diff
@@ -1551,8 +1564,76 @@ contract EVMRuntime is EVMConstants {
         state.errno = ERROR_INSTRUCTION_NOT_SUPPORTED;
     }
 
+    struct StackForCall {
+        uint gas;
+        address target;
+        uint value;
+        uint inOffset;
+        uint inSize;
+        uint retOffset;
+        uint retSize;
+    }
+
     function handleCALL(EVM memory state) internal {
+
+      StackForCall memory stack;
+      stack.gas = state.stack.pop();
+      stack.target = address(state.stack.pop());
+      stack.value = state.stack.pop();
+      stack.inOffset = state.stack.pop();
+      stack.inSize = state.stack.pop();
+      stack.retOffset = state.stack.pop();
+      stack.retSize = state.stack.pop();
+
+      uint gasFee = computeGasForMemory(state, stack.retOffset + stack.retSize, stack.inOffset + stack.inSize);
+
+      if (gasFee > state.gas) {
+          state.gas = 0;
+          state.errno = ERROR_OUT_OF_GAS;
+          return;
+      }
+      state.gas -= gasFee;
+
+      bytes memory cd = state.mem.toBytes(stack.inOffset, stack.inSize);
```

Comment:
maybe use `cData` or `callDataBytes` here, to keep it readable

Reply:
Done.